### PR TITLE
Fixed AsyncBackendJobExecutionActor's NullPointerExceptions

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Comment by cjllanwarne** [comment]
Any 2 for the review please :)

**Comment by Horneth** [comment]
the `context` is also not accessible if not in the `receive` thread ?

**Comment by Horneth** [comment]
👍

File: backend/src/main/scala/cromwell/backend/async/AsyncBackendJobExecutionActor.scala [hunk 1]
```diff
@@ -48,38 +48,36 @@ trait AsyncBackendJobExecutionActor { this: Actor with ActorLogging =>
     * If the `work` `Future` completes successfully, perform the `onSuccess` work, otherwise schedule
     * the execution of the `onFailure` work using an exponential backoff.
     */
-  def withRetry(work: Future[ExecutionHandle], onSuccess: ExecutionHandle => Unit, onFailure: => Unit): Unit = {
+  def withRetry(actorContext: ActorContext, work: Future[ExecutionHandle], onSuccess: ExecutionHandle => Unit, onFailure: => Unit)(implicit ec: ExecutionContext): Unit = {
```
Comment (geoffjentry) [inline backend/src/main/scala/cromwell/backend/async/AsyncBackendJobExecutionActor.scala]: Why is this not using the real retry?

I know the answer is "cuz it's already there" but if we're fixing it anyways ...

Comment (geoffjentry) [inline backend/src/main/scala/cromwell/backend/async/AsyncBackendJobExecutionActor.scala]: I opened #1004

Comment (cjllanwarne) [inline backend/src/main/scala/cromwell/backend/async/AsyncBackendJobExecutionActor.scala]: If I was personally going to rationalise all this async stuff, it wouldn't be by trying to crowbar it into yet-another-`retry`-method.

The whole retry logic here just seems philosophically anti-actor to me?

**Comment by geoffjentry** [comment]
👍

**Pull request merged**